data/CMakeLists.txt

```diff
@@ -0,0 +1,118 @@
+list(APPEND DATA_FILES
+  amr-hex.mesh
+  amr-quad.mesh
+  ball-nurbs.mesh
+  beam-hex-nurbs.mesh
+  beam-hex.mesh
+  beam-hex.vtk
+  beam-quad-amr.mesh
+  beam-quad-nurbs-sf.mesh
+  beam-quad-nurbs.mesh
+  beam-quad.mesh
+  beam-quad.vtk
+  beam-tet.mesh
+  beam-tet.vtk
+  beam-tri.mesh
+  beam-tri.vtk
+  beam-wedge.mesh
+  beam-wedge.vtk
+  compass.geo
+  compass.mesh
+  compass.msh
+  cube-nurbs.mesh
+  diag-segment-2d.mesh
+  diag-segment-3d.mesh
+  disc-nurbs.mesh
+  escher-p2.mesh
+  escher-p2.vtk
+  escher-p3.mesh
+  escher.mesh
+  escher.vtk
+  fichera-amr.mesh
+  fichera-mixed-p2.mesh
+  fichera-mixed-p2.vtk
+  fichera-mixed.mesh
+  fichera-q2.mesh
+  fichera-q2.vtk
+  fichera-q3.mesh
+  fichera-quad-mixed.mesh
+  fichera-quad.mesh
+  fichera.mesh
+  fichera.vtk
+  hexagon.mesh
+  inline-hex.mesh
+  inline-pyramid.mesh
+  inline-quad.mesh
+  inline-segment.mesh
+  inline-tet.mesh
+  inline-tri.mesh
+  inline-wedge.mesh
+  klein-bottle.mesh
+  klein-donut.mesh
+  l-shape.mesh
+  llnl-p3.mesh
+  mobius-strip.mesh
+  octahedron.mesh
+  periodic-annulus-sector.geo
+  periodic-annulus-sector.msh
+  periodic-cube.geo
+  periodic-cube.mesh
+  periodic-cube.msh
+  periodic-hexagon.mesh
+  periodic-segment.mesh
+  periodic-square.geo
+  periodic-square.mesh
+  periodic-square.msh
+  periodic-torus-sector.geo
+  periodic-torus-sector.msh
+  pipe-nurbs-2d.mesh
+  pipe-nurbs-log.mesh
+  pipe-nurbs.mesh
+  ref-cube.mesh
+  ref-prism.mesh
+  ref-pyramid.mesh
+  ref-segment.mesh
+  ref-square.mesh
+  ref-tetrahedron.mesh
+  ref-triangle.mesh
+  rt-2d-p4-tri.mesh
+  rt-2d-q3.mesh
+  segment-nurbs.mesh
+  square-disc-nurbs-patch.mesh
+  square-disc-nurbs.mesh
+  square-disc-p2.mesh
+  square-disc-p2.vtk
+  square-disc-p3.mesh
+  square-disc-surf.mesh
+  square-disc.mesh
+  square-disc.vtk
+  square-mixed.mesh
+  square-nurbs-pw.mesh
+  square-nurbs.mesh
+  star-hilbert.mesh
+  star-mixed-p2.mesh
+  star-mixed-p2.vtk
+  star-mixed.mesh
+  star-q2.mesh
+  star-q2.vtk
+  star-q3.fms
+  star-q3.mesh
+  star-surf.mesh
+  star.mesh
+  star.vtk
+  tinyzoo-3d.mesh
+  toroid-hex.mesh
+  toroid-wedge.mesh
+)
+
+foreach(DATA_FILE ${DATA_FILES})
```

Comment:
```suggestion
file(GLOB DATA_FILES ${CMAKE_CURRENT_SOURCE_DIR}/*)

foreach(DATA_PATH ${DATA_FILES})
  get_filename_component(DATA_FILE ${DATA_PATH} NAME)
```

Reply:
Right, I did try something similar but abandoned it because the reason you highlighted (which is why I think such loops are avoided in source directories in general).

Reply:
I see, I've edited my suggestion above to trigger a reconfiguration if the `data` source directory is modified (which happens if you create a new file).

Reply:
Thanks @nmnobre , this looks promising: I'll test it out.

Reply:
This checks all the boxes, thanks @nmnobre! I have minor syntax things to tweak, so I implemented it in [9672921](https://github.com/mfem/mfem/pull/4496/commits/96729213ef37b69b6f8273e2d8e87f00e0ac4f20) instead of applying this suggestion directly. As such, I'll leave it to you to officially resolve this thread if there isn't more discussion.